### DISCUSSION TITLE
Kubernates clustering

### DISCUSSION THREAD

**Discussion by Wiedzmin89** [OvenMediaLabs/OvenMediaEngine#1619] [Help]
Is there a way to run origins and edges on kubernates cluster with deployment scaling? Official documentation doesn't provide any info for kubernates clustering.

Can anyone provide Helm chart or at least .yml config for kubectl ?

**Comment by d-uzlov** [comment]
I believe that OME clustering for k8s isn't much different than OME clustering without k8s. And individual deployments aren't much different than deployment configs for any other app.
You deploy 1+ origins, 1 redis, 1 or several edges¹, point them to each other, same as without k8s.

Here is an example of how I do it in my k8s cluster: https://github.com/d-uzlov/k8s-homelab/tree/master/video/ome
Not sure how useful it would be for you.

¹ One k8s-specific pitfall that I encountered is that you can't easily scale edge, at least when using WebRTC, because if there are several edge instances behind a single load-balancer, and webtrc data request goes to a different instance than the control plane request, things seems to break.
I think, the only way to reliably scale the edge is to define several load-balancers, one for each instance.
You also probably need to use StatefulSet for edge in this case, to be able to distinguish instances. Or just different deployments.
I didn't try it too hard, I ended up using just a single edge instance.
